src/Build/Logging/TerminalLogger/TerminalLogger.cs

```diff
@@ -599,28 +621,34 @@ private void StatusEventRaised(object sender, BuildStatusEventArgs e)
     /// </summary>
     private void ProjectStarted(object sender, ProjectStartedEventArgs e)
     {
-        var buildEventContext = e.BuildEventContext;
-        if (buildEventContext is null)
+        if (e.BuildEventContext is null)
         {
             return;
         }
 
-        ProjectContext c = new ProjectContext(buildEventContext);
+        ProjectContext c = new(e.BuildEventContext);
 
         if (_restoreContext is null)
         {
-            if (e.GlobalProperties?.TryGetValue("TargetFramework", out string? targetFramework) != true)
+            EvalContext evalContext = new(e.BuildEventContext);
+            string? targetFramework = null;
+            string? runtimeIdentifier = null;
+            if (_evals.TryGetValue(evalContext, out EvalProjectInfo evalInfo))
             {
-                targetFramework = null;
+                targetFramework = evalInfo.TargetFramework;
+                runtimeIdentifier = evalInfo.RuntimeIdentifier;
             }
-            _projects[c] = new(e.ProjectFile!, targetFramework, CreateStopwatch?.Invoke());
+            System.Diagnostics.Debug.Assert(evalInfo != default, "EvalProjectInfo should have been captured before ProjectStarted");
+
+            TerminalProjectInfo projectInfo = new(c, evalInfo, CreateStopwatch?.Invoke());
+            _projects[c] = projectInfo;
```

Comment:
The main difference here is that we now just have to look up the eval context when a project starts building, instead of trying to parse the properties from the ProjectStarted event.